packages/jsii/lib/assembler.ts

```diff
@@ -479,11 +505,47 @@ export class Assembler implements Emitter {
         return _sortMembers(this._visitDocumentation(type.symbol, jsiiType));
     }
 
-    private async _visitEnum(type: ts.Type, namespace: string[]): Promise<spec.EnumType> {
+    /**
+     * @returns true if this member is internal and should be omitted from the type manifest
+     */
+    private _isPrivateOrInternal(symbol: ts.Symbol, validateDeclaration?: ts.Declaration): boolean {
+        const hasInternalJsDocTag = _hasInternalJsDocTag(symbol);
+        const hasUnderscorePrefix = symbol.name !== '__constructor' && symbol.name.startsWith('_');
+
+        if (_isPrivate(symbol)) {
+            LOG.trace(`skipping ${symbol.name} because it is marked "private"`);
+            return true;
+        }
+
+        if (!hasInternalJsDocTag && !hasUnderscorePrefix) {
+            return false;
+        }
+
+        // we only validate if we have a declaration
+        if (validateDeclaration) {
+            if (!hasUnderscorePrefix) {
+                this._diagnostic(validateDeclaration, ts.DiagnosticCategory.Error,
+                    `${symbol.name}: the name of members marked as @internal must begin with an underscore`);
```

Comment:
I don't know if you need to mention `symbol.name` here, since you're attaching the diagnostic to the declaration.

Reply:
From my experience with the errors this was helpful